packages/mdc-list/foundation.js

```diff
@@ -371,6 +412,38 @@ class MDCListFoundation extends MDCFoundation {
   hasCheckboxOrRadioAtIndex_(index) {
     return this.adapter_.hasCheckboxAtIndex(index) || this.adapter_.hasRadioAtIndex(index);
   }
+
+  /**
+   * @param {!Index} index
+   * @return {boolean}
+   */
+  isIndexValid_(index) {
+    const listSize = this.adapter_.getListItemCount();
+
+    if (index instanceof Array) {
+      if (!this.hasCheckboxAtIndex_(index)) {
+        throw new Error('MDCListFoundation: Array of index is only supported for checkbox based list');
+      }
+      return index.some((i) => this.isIndexInRange_(i));
+    } else {
+      return index >= 0 && index < listSize;
```

Comment:
Could this also use `this.isIndexInRange_`? (See also my comment in that method - currently this code path will never throw an error, but that code path will.)